### PR TITLE
model: h2oautoml: Add h2o AutoML

### PR DESCRIPTION
**Pull request by nitesh585**
#959

### CODE REVIEW

File: model/h2oautoml/.github/workflows/testing.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+name: Tests
```
Comment (pdxjohnny) [inline model/h2oautoml/.github/workflows/testing.yml:1]: Can remove this file

File: model/h2oautoml/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# DFFML model_name Models
```
Comment (pdxjohnny) [inline model/h2oautoml/README.md:1]: Update this

File: model/h2oautoml/dffml_model_h2oautoml/h2oautoml.py [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+            self.logger.info(str(self.saved.leaderboard))
+
+        # get model ids from leader_board of h2o's Automl
+        leader_board = self.saved.leaderboard
+        model_ids = list(leader_board["model_id"].as_data_frame().iloc[:, 0])
+
+        # get leader model of the leader_board
+        mdl = h2o.get_model(model_ids[0])
+
+        # save the leader model
```
Comment (pdxjohnny) [inline model/h2oautoml/dffml_model_h2oautoml/h2oautoml.py:271]: nit: Capitalize first letter of comments, for consistency

File: model/h2oautoml/dffml_model_h2oautoml/h2oautoml.py [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+        """
+
+        xdata = []
+        ydata = []
+        async for record in sources.with_features(
+            self.features + [self.parent.config.predict.name]
+        ):
+            record_data = []
+            for feature in record.features(self.features).values():
+                record_data.extend(
```
Comment (pdxjohnny) [inline model/h2oautoml/dffml_model_h2oautoml/h2oautoml.py:220]: Why do we convert features to a list if they are a scalar?

File: model/h2oautoml/setup.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    "setup_common", os.path.join(os.path.dirname(__file__), "setup_common.py")
+)
+common = importlib.util.module_from_spec(spec)
+spec.loader.exec_module(common)
+
+common.KWARGS["install_requires"] += ["requests>=2.25.1"]
+common.KWARGS["install_requires"] += ["colorama>=0.3.8"]
+common.KWARGS["install_requires"] += ["tabulate>=0.8.7"]
+common.KWARGS["install_requires"] += ["future>=0.18.2"]
+common.KWARGS["install_requires"] += [" h2o>=3.32.0.3"]
```
Comment (pdxjohnny) [inline model/h2oautoml/setup.py:17]:
```suggestion
common.KWARGS["install_requires"] += ["h2o>=3.32.0.3"]
```

File: model/h2oautoml/.gitignore [hunk 6]
```diff
@@ -0,0 +1,20 @@
+*.log
```
Comment (pdxjohnny) [inline model/h2oautoml/.gitignore:1]: I think we can remove this file

**Review by pdxjohnny — CHANGES_REQUESTED** [review]
We'll wait on #979 before merging this. We'll need to update this and lightgbm to the setup.cfg style after that is merged.

File: model/h2oautoml/dffml_model_h2oautoml/h2oautoml.py [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+        self.saved_filepath = pathlib.Path(self.config.directory)
+
+        # Load saved model if it exists
+        try:
+            path = (
+                str(self.saved_filepath)
+                + "/"
+                + str(os.listdir(str(self.saved_filepath))[1])
+            )
+            print("\n\n", path)
```
Comment (pdxjohnny) [inline model/h2oautoml/dffml_model_h2oautoml/h2oautoml.py:203]: Do we really want to print?

File: model/h2oautoml/dffml_model_h2oautoml/h2oautoml.py [hunk 8]
```diff
@@ -0,0 +1,337 @@
...
+            self.saved = h2o.load_model(path)
+        except:
+            pass
```
Comment (pdxjohnny) [inline model/h2oautoml/dffml_model_h2oautoml/h2oautoml.py:206]: We should probably log on error here (self.logger.error or something)

Comment (pdxjohnny) [reply model/h2oautoml/dffml_model_h2oautoml/h2oautoml.py:206]: Can we check if the path exists first or does that not help us here?

Comment (pdxjohnny) [reply model/h2oautoml/dffml_model_h2oautoml/h2oautoml.py:206]: @nitesh585 I wanted to get this merged so it doesn't get stale, can you open another PR to fix these small things please?

Comment (pdxjohnny) [reply model/h2oautoml/dffml_model_h2oautoml/h2oautoml.py:206]: That was a mistake, I should not have merged this, I backed it out the tests weren't setup to run in CI and then failed: https://github.com/pdxjohnny/dffml/tree/h2oautoml

Comment (pdxjohnny) [reply model/h2oautoml/dffml_model_h2oautoml/h2oautoml.py:206]: Please open another PR

Comment (nitesh585) [reply model/h2oautoml/dffml_model_h2oautoml/h2oautoml.py:206]: @pdxjohnny  sure! I will open another PR.

File: model/h2oautoml/dffml_model_h2oautoml/h2oautoml.py (hunk 7 above)
Comment (nitesh585) [reply model/h2oautoml/dffml_model_h2oautoml/h2oautoml.py:203]: No, its just a for debuggin. I have removed it.

**Pull request merged**